cmd/yggd/main.go

```diff
@@ -322,23 +333,28 @@ func main() {
 		if err != nil {
 			return cli.Exit(fmt.Errorf("cannot read contents of directory: %w", err), 1)
 		}
-		configDir := filepath.Join(yggdrasil.SysconfDir, yggdrasil.LongName)
-		env := []string{
-			"YGG_SOCKET_ADDR=unix:" + c.String("socket-addr"),
-			"PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
-			"YGG_CONFIG_DIR=" + configDir,
-			"YGG_LOG_LEVEL=" + level.String(),
-			"YGG_CLIENT_ID=" + ClientID,
-		}
 		for _, info := range fileInfos {
-			if strings.HasSuffix(info.Name(), "worker") {
-				log.Debugf("starting worker: %v", info.Name())
-				go startProcess(filepath.Join(workerPath, info.Name()), env, 0, d.deadWorkers)
+			log.Debugf("starting worker: %v", info.Name())
+			config, err := loadWorkerConfig(filepath.Join(workerPath, info.Name()))
+			if err != nil {
+				log.Errorf("cannot load worker config: %v", err)
+				continue
 			}
+			if ExcludeWorkers[config.directive] {
+				continue
+			}
```

Comment:
It would be better to move this test before: 

`log.Debugf("starting worker: %v:", )`

Otherwise, we get a that debug message even if the worker has been excluded.

Reply:
Yea that makes sense.